### PR TITLE
[RFC] boards: msp432p401r: Add GPIO Support

### PR DESCRIPTION
**Pull request by DhruvaG2000**
Adds GPIO Support for the TI MSP432P4012R MCU

Signed-off-by: Dhruva Gole <goledhruva@gmail.com>

### CODE REVIEW

**Comment by DhruvaG2000** [comment]
This PR Doesn't build properly yet, need help on what changes need to be added exactly for adding gpio support to this board other than the gpio node in dts.

File: dts/arm/ti/msp432p4xx.dtsi [hunk 1]
```diff
@@ -1,6 +1,9 @@
 /* SPDX-License-Identifier: Apache-2.0 */
 
 #include <arm/armv7-m.dtsi>
+#include <zephyr/dt-bindings/gpio/gpio.h>
+
+#define EXC_GPIOA0  0  /* (INT_GPIOA0 - 16) = (16-16) */
```
Comment (vaishnavachath) [inline dts/arm/ti/msp432p4xx.dtsi:7]: @DhruvaG2000 where do you use this define?

File: dts/arm/ti/msp432p4xx.dtsi [hunk 2]
```diff
@@ -35,6 +38,13 @@
 			clocks = <&sysclk>;
 			status = "disabled";
 		};
+		gpioa: gpio@40005000 {
+			compatible = "ti,msp432p4xx-gpio";
+			reg = <0x40005000 0x1000>;
```
Comment (vaishnavachath) [inline dts/arm/ti/msp432p4xx.dtsi:43]: Does the GPIO driver exist for this platform? I couldn't find one, where did you get this compatible from?
you might need to add something around https://github.com/zephyrproject-rtos/hal_ti/blob/master/simplelink/source/ti/devices/msp432p4xx/driverlib/gpio.c

**Review by vaishnavachath — CHANGES_REQUESTED** [review]
@DhruvaG2000 refer drivers/gpio/gpio_cc13xx_cc26xx.c and https://github.com/zephyrproject-rtos/hal_ti/blob/master/simplelink/source/ti/devices/cc13x2_cc26x2/driverlib/gpio.c then you will get an idea of what needs to be done to enable GPIO support for this platform.

File: dts/arm/ti/msp432p4xx.dtsi (hunk 2 above)
Comment (DhruvaG2000) [reply dts/arm/ti/msp432p4xx.dtsi:43]: Thanks for your inputs vaishnav, I have started writing a GPIO driver based on your suggestions.

File: drivers/gpio/gpio_msp432p4xx.c [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+#include <inc/hw_types.h>
+#include <inc/hw_memmap.h>
+#include <inc/hw_ints.h>
+#include <inc/hw_gpio.h>
+#include <driverlib/rom.h>
+#include <driverlib/pin.h>
+
+#include "gpio_utils.h"
+
+static const uint8_t pinTable[] = {
```
Comment (DhruvaG2000) [inline drivers/gpio/gpio_msp432p4xx.c:26]: @vaishnavachath  can you help me with where these values are supposed to come from? What do they mean?

**Pull request closed without merging**